reader.go

```diff
@@ -53,6 +75,18 @@ func Open(filename string) (*Reader, error) {
 	return s, nil
 }
 
+// Open opens a Shapefile for reading using memory buffers
+func OpenFromMemory(shpFileData, dbfFileData []byte) (*Reader, error) {
+	mD := &memoryShapeData{
+		shpFileData: shpFileData,
+		dbfFileData: dbfFileData,
+	}
+
+	s := &Reader{filename: "", shp: newBytesReader(mD.shpFileData), memoryData: mD}
```

Comment:
I have doubts about this, as well. Effectively, you are adding references to `mD` twice to `s`.  I understand that you want to preserve the dbf data so that openDbf has something to open later, but the proposed design will overcomplicate things.